<a href="https://colab.research.google.com/github/Marc86316/DataCamp/blob/main/Competition/Demystifying_data_salaries__Level_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## 📖 Background
You work for an international HR consultancy helping companies attract and retain top talent in the competitive tech industry. As part of your services, you provide clients with insights into industry salary trends to ensure they remain competitive in hiring and compensation practices.

Your team wants to use a data-driven approach to analyse how various factors—such as job role, experience level, remote work, and company size—impact salaries globally. By understanding these trends, you can advise clients on offering competitive packages to attract the best talent.

In this competition, you’ll explore and visualise salary data from thousands of employees worldwide. f you're tackling the advanced level, you'll go a step further—building predictive models to uncover key salary drivers and providing insights on how to enhance future data collection.

## 💾 The data

The data comes from a survey hosted by an HR consultancy, available in `'salaries.csv'`.

#### Each row represents a single employee's salary record for a given year:
- **`work_year`** - The year the salary was paid.  
- **`experience_level`** - Employee experience level:  
  - **`EN`**: Entry-level / Junior  
  - **`MI`**: Mid-level / Intermediate  
  - **`SE`**: Senior / Expert  
  - **`EX`**: Executive / Director  
- **`employment_type`** - Employment type:  
  - **`PT`**: Part-time  
  - **`FT`**: Full-time  
  - **`CT`**: Contract  
  - **`FL`**: Freelance  
- **`job_title`** - The job title during the year.  
- **`salary`** - Gross salary paid (in local currency).  
- **`salary_currency`** - Salary currency (ISO 4217 code).  
- **`salary_in_usd`** - Salary converted to USD using average yearly FX rate.  
- **`employee_residence`** - Employee's primary country of residence (ISO 3166 code).  
- **`remote_ratio`** - Percentage of remote work:  
  - **`0`**: No remote work (<20%)  
  - **`50`**: Hybrid (50%)  
  - **`100`**: Fully remote (>80%)  
- **`company_location`** - Employer's main office location (ISO 3166 code).  
- **`company_size`** - Company size:  
  - **`S`**: Small (<50 employees)  
  - **`M`**: Medium (50–250 employees)  
  - **`L`**: Large (>250 employees)  

In [ ]:
import pandas as pd
salaries_df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/DataCamp/Competition/Demystifying data salaries 💰 - Level 1/salaries.csv')
salaries_df

work_year experience_level employment_type                 job_title  \
0           2024               MI              FT                 Developer   
1           2024               MI              FT                 Developer   
2           2024               EN              FT                 Developer   
3           2024               EN              FT                 Developer   
4           2024               EN              FT                 Developer   
...          ...              ...             ...                       ...   
57189       2020               SE              FT            Data Scientist   
57190       2021               MI              FT  Principal Data Scientist   
57191       2020               EN              FT            Data Scientist   
57192       2020               EN              CT     Business Data Analyst   
57193       2021               SE              FT            Data Scientist   

        salary salary_currency  salary_in_usd employee_residence  \
0       168276             USD         168276                 US   
1       112184             USD         112184                 US   
2       180000             USD         180000                 US   
3       133500             USD         133500                 US   
4       122000             USD         122000                 US   
...        ...             ...            ...                ...   
57189   412000             USD         412000                 US   
57190   151000             USD         151000                 US   
57191   105000             USD         105000                 US   
57192   100000             USD         100000                 US   
57193  7000000             INR          94665                 IN   

       remote_ratio company_location company_size  
0                 0               US            M  
1                 0               US            M  
2                 0               US            M  
3                 0               US            M  
4                 0               US            M  
...             ...              ...          ...  
57189           100               US            L  
57190           100               US            L  
57191           100               US            S  
57192           100               US            L  
57193            50               IN            L  

[57194 rows x 11 columns]

## 💪 Competition challenge

In this first level, you’ll explore and summarise the dataset to understand its structure and key statistics. If you want to push yourself further, check out level two!
Create a report that answers the following:
- How many records are in the dataset, and what is the range of years covered?
- What is the average salary (in USD) for Data Scientists and Data Engineers? Which role earns more on average?
- How many full-time employees based in the US work 100% remotely?

## 🧑‍⚖️ Judging criteria

This is a community-based competition. Once the competition concludes, you'll have the opportunity to view and vote for the best submissions of others as the voting begins. The top 5 most upvoted entries will win. The winners will receive DataCamp merchandise.

## ✅ Checklist before publishing into the competition
- Rename your workspace to make it descriptive of your work. N.B. you should leave the notebook name as notebook.ipynb.
- **Remove redundant cells** like the judging criteria, so the workbook is focused on your story.
- Make sure the workbook reads well and explains how you found your insights.
- Try to include an **executive summary** of your recommendations at the beginning.
- Check that all the cells run without error

## ⌛️ Time is ticking. Good luck!

# Task
In this first level, you’ll explore and summarise the dataset to understand its structure and key statistics. If you want to push yourself further, check out level two! Create a report that answers the following:

How many records are in the dataset, and what is the range of years covered?
What is the average salary (in USD) for Data Scientists and Data Engineers? Which role earns more on average?
How many full-time employees based in the US work 100% remotely?

Here is all the data you need:
"salaries.csv"

## Data loading

### Subtask:
Load the "salaries.csv" file into a pandas DataFrame.


**Reasoning**:
Load the "salaries.csv" file into a pandas DataFrame.



In [ ]:
import pandas as pd

df = pd.read_csv('salaries.csv')
display(df.head())

work_year experience_level employment_type  job_title  salary  \
0       2024               MI              FT  Developer  168276   
1       2024               MI              FT  Developer  112184   
2       2024               EN              FT  Developer  180000   
3       2024               EN              FT  Developer  133500   
4       2024               EN              FT  Developer  122000   

  salary_currency  salary_in_usd employee_residence  remote_ratio  \
0             USD         168276                 US             0   
1             USD         112184                 US             0   
2             USD         180000                 US             0   
3             USD         133500                 US             0   
4             USD         122000                 US             0   

  company_location company_size  
0               US            M  
1               US            M  
2               US            M  
3               US            M  
4               US            M

## Data exploration

### Subtask:
Explore the dataset to understand its basic characteristics, specifically the number of records and the range of years covered.


**Reasoning**:
I need to determine the number of records and the range of years covered in the dataset.



In [ ]:
# Determine the number of records
num_records = len(df)

# Find the minimum and maximum years
min_year = df['work_year'].min()
max_year = df['work_year'].max()

# Print the results (for verification)
print(f"Number of records: {num_records}")
print(f"Range of years: {min_year} to {max_year}")

Number of records: 57194
Range of years: 2020 to 2024


## Data analysis

### Subtask:
Calculate the average salary for Data Scientists and Data Engineers, and determine which role has a higher average salary.  Also, count the number of full-time employees based in the US who work 100% remotely.


**Reasoning**:
Calculate the average salaries for Data Scientists and Data Engineers, identify the higher-paying role, and count the number of full-time remote employees in the US.



In [ ]:
# Calculate average salaries
data_scientists = df[df['job_title'] == 'Data Scientist']
data_engineers = df[df['job_title'] == 'Data Engineer']

avg_data_scientist_salary = data_scientists['salary_in_usd'].mean()
avg_data_engineer_salary = data_engineers['salary_in_usd'].mean()

# Compare averages
if avg_data_scientist_salary > avg_data_engineer_salary:
    higher_paying_role = 'Data Scientist'
else:
    higher_paying_role = 'Data Engineer'

# Count remote workers
remote_workers = df[(df['employment_type'] == 'FT') &
                    (df['company_location'] == 'US') &
                    (df['remote_ratio'] == 100)]
num_remote_us_employees = len(remote_workers)

print(f"Average Data Scientist Salary: ${avg_data_scientist_salary:.2f}")
print(f"Average Data Engineer Salary: ${avg_data_engineer_salary:.2f}")
print(f"Higher Paying Role: {higher_paying_role}")
print(f"Number of remote US employees: {num_remote_us_employees}")

Average Data Scientist Salary: $159397.07
Average Data Engineer Salary: $149315.00
Higher Paying Role: Data Scientist
Number of remote US employees: 11160


## Summary:

### 1. Q&A
* **How many records are in the dataset, and what is the range of years covered?** The dataset contains 57,194 records and covers the years 2020 to 2024.
* **What is the average salary (in USD) for Data Scientists and Data Engineers? Which role earns more on average?**  The average salary for Data Scientists is $159,397.07, while the average salary for Data Engineers is $149,315.00. Data Scientists earn more on average.
* **How many full-time employees based in the US work 100% remotely?** There are 11,160 full-time employees based in the US who work 100% remotely.

### 2. Data Analysis Key Findings
* Data Scientists earn an average salary of $159,397.07, compared to $149,315.00 for Data Engineers.
* 11,160 full-time employees located in the US work 100% remotely.

### 3. Insights or Next Steps
* **Salary Discrepancy:** Investigate the factors contributing to the salary difference between Data Scientists and Data Engineers.  Further analysis could explore experience level, company size, or specific skills.
* **Remote Work Trend:** Analyze the remote work trend over the years (2020-2024) to see if there's a correlation with job title, location, or company size.


## Executive Summary
    
* This analysis of salary data reveals that Data Scientists earn more on average than Data Engineers.

* A significant number of full-time employees in the US are working 100% remotely.

* Further investigation is recommended to understand the factors contributing to the salary discrepancy between roles and to explore the trend of remote work over time.